aws/resource_aws_appmesh_mesh.go

```diff
@@ -127,6 +130,15 @@ func resourceAwsAppmeshMeshRead(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("error setting spec: %s", err)
 	}
 
+	if err := saveTagsAppmesh(conn, d, aws.StringValue(resp.Mesh.Metadata.Arn)); err != nil {
```

Comment:
I think the preferred style for error handling is

```go
err = saveTagsAppmesh(conn, d, aws.StringValue(resp.Mesh.Metadata.Arn))
if isAWSErr(err, appmesh.ErrCodeNotFoundException, "") {
    log.Printf("[WARN] App Mesh service mesh (%s) not found, removing from state", d.Id())
    d.SetId("")
    return nil
}
if err != nil {
    return fmt.Errorf("error saving tags: %s", err)
}
```

Reply:
Fixed.